src/main/scala/org/clulab/wm/discovery/ConceptDiscovery.scala

```diff
@@ -31,18 +35,41 @@ class ConceptDiscovery {
     val entityFinder = RuleBasedEntityFinder.fromConfig(config, tagSet, stopwordManager)
     val processor = new CluCoreProcessor()
     val conceptLocations = mutable.Map.empty[String, Set[String]].withDefaultValue(Set.empty)
-    for (path <- documents) {
-      val text = FileUtils.getTextFromFile(path.toString)
-      val document = processor.annotate(text)
+    for ((cdr, idx) <- cdrs.zipWithIndex) {
+      // make a Processors Document, pruning sentences with a threshold if applicable
+      val document = mkDocument(processor, cdr.sentences, sentenceThreshold)
       val mentions = entityFinder.find(document)
       val trimmed_mentions = mentions.map(EntityHelper.trimEntityEdges(_, tagSet))
       val annotatedDocument = AnnotatedDocument(document, trimmed_mentions)
       for (mention <- annotatedDocument.odinMentions) {
-        conceptLocations(mention.text) += s"${path}:${mention.sentence}"
+        conceptLocations(mention.text) += s"${cdr.docid}:${mention.sentence}"
       }
     }
-    Set.empty ++ conceptLocations.map {
-      case (phrase, documentLocations) => Concept(phrase, documentLocations)
+
+    conceptLocations.map{
+      case (phrase, locations) => Concept(phrase, locations)
+    }.toSet
+  }
+
+  /** Currently pruning sentences from the text whose Qntfy sentence score are below a threshold.
+    * TODO: should this be a top k instead? */
+  def mkDocument(processor: Processor, sentences: Seq[ScoredSentence], sentenceThreshold: Option[Double]): Document = {
+    if (sentenceThreshold.isEmpty) {
+      processor.mkDocumentFromSentences(sentences.map(_.text))
+    }
+    else {
+      // ranked in descending order of score
+      val rankedSentences = sentences.sortBy(-_.score)
+
+      // only accept sentences with a saliency above some threshold
+      val lastValid = rankedSentences.indexWhere(_.score < sentenceThreshold.get)
+      val validSentences = rankedSentences
+        .slice(0, lastValid)
+        // put them back in order
+        .sortBy(_.start)
+
+      // annotate and return as a Document
+      processor.annotateFromSentences(validSentences.map(_.text))
```

Comment:
@BeckySharp I think this whole method is equivalent to the following code:

    val validSentences = sentenceThreshold match {
      case None => sentences
      case Some(threshold) => sentences.filter(_.score >= threshold).sortBy(_.start)
    }
    // annotate and return as a Document
    processor.annotateFromSentences(validSentences.map(_.text))

I can make this change if that's true, but since there are no tests for this behavior, I need some confirmation that this is indeed the intent.

Reply:
Or maybe even more succinctly:

    val selectedSentencesOption = sentenceThreshold.map(t => sentences.filter(_.score >= t))
    processor.annotateFromSentences(selectedSentencesOption.getOrElse(sentences).map(_.text))